engine/src/flutter/shell/platform/linux/fl_view.cc

```diff
@@ -784,9 +792,12 @@ G_MODULE_EXPORT FlView* fl_view_new_for_engine(FlEngine* engine) {
 
   self->pointer_manager = fl_pointer_manager_new(self->view_id, engine);
 
+  // TODO(mattkae): We should make the fl_engine_set_update_semantics_handler
+  // that iterates over all of the views and finds the right one. At the moment,
+  // we only update semantics on the first view.
```

Comment:
This should already work - we removed the handler and used a signal so all the views subscribe to it. From `fl_engine.cc`:
```c
// Called when a semantic node update is received from the engine.
static void fl_engine_update_semantics_cb(const FlutterSemanticsUpdate2* update,
                                          void* user_data) {
  FlEngine* self = FL_ENGINE(user_data);

  g_signal_emit(self, fl_engine_signals[SIGNAL_UPDATE_SEMANTICS], 0, update);
}
```